granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -88,13 +198,46 @@ def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> O
     v2_path = find_v2_hierarchy(resolve_host_root_links)
     if v1_paths and controller in v1_paths:
         # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
-        return v1_paths[controller]
+        return CgroupCoreV1(Path(v1_paths[controller]))
     if v2_path:
-        # v2 (unified) - all controllers are in a single unified hierarchy
-        return v2_path
+        # v2 (unified) - check given controller is supported in cgroup v2
+        cgroup_v2 = CgroupCoreV2(Path(v2_path))
+        if cgroup_v2.is_controller_supported(controller):
```

Comment:
This will happen if the particular controller is already controlled by the parent cgroup, right? In that case I'd say we want a specific exception and NOT a behavior of "controller is not available", which is implied by this function returning None.

Reply:
Also, the comment below `# No cgroup mount of the requested controller` is misleading because that's not the case here.

Reply:
I see it's checked in `get_cgroup_mount_checked`, commenting there as well

Reply:
Specifically here, `cgroup_v2.is_controller_supported(controller)` is checking the controller is supported on the mount point of cgroup v2, which might be False if this controller isn't supported for v2 (i.e. the specific controller is not mounted for either v1 or v2)

Reply:
You are right that the usage of `is_controller_supported` in other contexts in the code, might return False because it's being used in a parent cgroup.